sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -794,14 +794,29 @@ export class EventHubReceiver extends LinkEntity {
     };
 
     const retryOptions = this.options.retryOptions || {};
-    const config: RetryConfig<ReceivedEventData[]> = {
-      connectionHost: this._context.config.host,
-      connectionId: this._context.connectionId,
-      operation: retrieveEvents,
-      operationType: RetryOperationType.receiveMessage,
-      abortSignal: abortSignal,
-      retryOptions: retryOptions
-    };
+
+    const config: RetryConfig<ReceivedEventData[]> = Object.defineProperties(
+      {
+        operation: retrieveEvents,
+        operationType: RetryOperationType.receiveMessage,
+        abortSignal: abortSignal,
+        retryOptions: retryOptions
+      },
+      {
+        connectionId: {
+          enumerable: true,
+          get: () => {
```

Comment:
This is very cool and makes me think we should always make these properties dynamic (ie, Service Bus too).

In a separate PR I was thinking it'd be nice to just pass in a logger (and maybe it's time to make it so the logger properly prints out the header).

Reply:
I don't follow how the change here helps us... Was there something in the PR description I missed?

Reply:
Ah sorry, I didn't mention this in the PR. Basically, before this change, sometimes in our logs we'd see the wrong connection-id when referring to a receiver because we copied the value of the id. With this change, we're always referencing the id from our context object. Now our logs are accurate.

Reply:
I noticed this while trying to troubleshoot the too many pending tasks issue.

Reply:
Wowza, should we follow this format everywhere else we log the connection id? Or only certain scenarios need this? If yes for either question, can you log an issue to investigate all other places where we may have to change?

Reply:
We should look into following this method anywhere we would copy the connectionId value today. Issue created:
https://github.com/Azure/azure-sdk-for-js/issues/14923